internal/resource/client_service.go

```diff
@@ -56,13 +58,51 @@ func (builder *ClientServiceBuilder) Update(object runtime.Object) error {
 		}
 	}
 
+	if builder.Instance.Spec.Override.ClientService != nil {
+		if err := builder.applySvcOverride(service, builder.Instance.Spec.Override.ClientService); err != nil {
+			return fmt.Errorf("failed applying Client Service override: %v", err)
+		}
+	}
+
 	if err := controllerutil.SetControllerReference(builder.Instance, service, builder.Scheme); err != nil {
 		return fmt.Errorf("failed setting controller reference: %v", err)
 	}
 
 	return nil
 }
 
+func (builder *ClientServiceBuilder) applySvcOverride(svc *corev1.Service, override *rabbitmqv1beta1.ClientService) error {
```

Comment:
Does this need to be a pointer function? I'm not sure if it makes much of a difference in terms of performance or "correctness"?

What do you think @j4mcs ?

Reply:
ah it does not need to be, I will update the function signature.